stl/inc/utility

```diff
@@ -136,6 +136,28 @@ template <class _Ty, int _Enabled /* = 0 */>
 #endif // ^^^ !_HAS_CXX17 ^^^
 _CONSTEXPR20 void swap(_Ty& _Left, _Ty& _Right)
     noexcept(is_nothrow_move_constructible_v<_Ty> && is_nothrow_move_assignable_v<_Ty>) {
+    if (_STD addressof(_Left) == _STD addressof(_Right)) {
+        return; // Handle self-swap as a no-op; see LWG-4165
+    }
+
+#if defined(_M_IX86) || defined(_M_X64)
+    if constexpr (_STD is_integral_v<_Ty>) {
+        __asm {
+            mov eax, [&_Left]
+            xchg eax, [&_Right]
+            mov [&_Left], eax
+        }
+        return;
+    }
+#elif _ENABLE_XOR_SWAP
+    if constexpr (_STD is_integral_v<_Ty>) {
+        _Left ^= _Right;
+        _Right ^= _Left;
+        _Left ^= _Right;
+        return;
+    }
```

Comment:
I think this is pessimization - such pattern usually causes more asm instructions.

Reply:
It is correct, lots of extra instructions:
compare /O2 ^=
https://godbolt.org/z/vKroa85vd
with asm:
https://godbolt.org/z/zYMnah9z6

Reply:
`/O3` isn't even an existing option. Also agree with @frederick-vs-ja